### DISCUSSION TITLE
Browser redirect to sign up instead of log in?

### DISCUSSION THREAD

**Discussion by padraigpearlman** [fief-dev/fief#134] [Q&A] [answered]
I'm using Fief for a browser frontend (with a backend RESTful API). My landing page for the frontend website has two buttons: one to sign up and one to log in.

```
const loginButton = ...
const fiefAuth = new fief.browser.FiefAuth(fiefClient);

loginButton.addEventListener("click", () => {
    fiefAuth.redirectToLogin("my callback address");
});
```

There is no equivalent `redirectToSignUp` available. Is there a way to have a sign up button on my homepage that links to the sign up page instead of the log in page?

**Comment by fief-bailiff** [comment]
Hail, @pap024 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment, marked as answer]
Hi @pap024 👋 Great question! It's actually possible but was poorly documented and tooled. This is now fixed!

So here is how you can do that:

* We support an optional parameter `screen` on the authorize URL to force either the login or register page. See here ➡️ https://docs.fief.dev/going-further/authorize-url/
* The `redirectToLogin` method of the browser integration was not able to accept those optional parameters. This is now fixed in [`v0.12.0`](https://github.com/fief-dev/fief-js/releases/tag/v0.12.0). Example:

```ts
fiefAuth.redirectToLogin('http://localhost:8080/callback.html', { extrasParams: { screen: 'register' } });
```

**Reply by padraigpearlman** [reply]
Thank you!
